src/deploymentModals.jsx

```diff
@@ -0,0 +1,213 @@
+    const [deleteRPMmetadata, setDeleteRPMmetadata] = useState(true);
+    const [deletePendingDeployments, setDeletePendingDeployments] = useState(false);
+    const [deleteRollbackDeployments, setDeleteRollbackDeployments] = useState(false);
+    const [buttonLoading, setButtonLoading] = useState(false);
+    const [error, setError] = useState("");
+
+    const doCleanup = () => {
+        setButtonLoading(true);
+        setError("");
+
+        const cleanupFlags = [];
+        if (deleteTemporaryFiles) {
+            cleanupFlags.push("base");
+        }
+        if (deleteRPMmetadata) {
+            cleanupFlags.push("repomd");
+        }
+        if (deletePendingDeployments) {
+            cleanupFlags.push("pending-deploy");
+        }
+        if (deleteRollbackDeployments) {
+            cleanupFlags.push("rollback-deploy");
+        }
+
+        return client.run_transaction("Cleanup", [cleanupFlags], os)
+                .then(Dialogs.close)
+                .catch(ex => {
+                    console.warn(ex);
+                    setError(ex.message);
+                    setButtonLoading(false);
+                });
+    };
+
+    const actions = [
+        <Button key="cleanup-accept"
+            variant="primary"
+            isDisabled={buttonLoading || (!deleteTemporaryFiles && !deleteRPMmetadata && !deletePendingDeployments && !deleteRollbackDeployments)}
+            isLoading={buttonLoading}
+            onClick={() => doCleanup()}
+        >
+            {_("Clean up")}
+        </Button>,
+        <Button key="cleanup-cancel" variant="link" onClick={Dialogs.close}>
+            {_("Cancel")}
+        </Button>
+    ];
+
+    return (
+        <Modal isOpen
+            id="cleanup-deployment-modal"
+            title={_("Remove temporary files")}
+            position="top"
+            variant="small"
+            onClose={Dialogs.close}
+            actions={actions}
+        >
+            {error &&
+                <Alert variant="danger"
+                    isInline
+                    title={error}
+                />
+            }
+            <Stack>
+                <Checkbox label={_("Temporary files")}
+                    key="temporary-files"
+                    id="temporary-files-checkbox"
+                    onChange={(_, isChecked) => setDeleteTemporaryFiles(isChecked)}
+                    isChecked={deleteTemporaryFiles}
+                />
+                <Checkbox label={_("RPM repo metadata")}
+                    key="rpm-repo-metadata"
+                    id="rpm-repo-metadata-checkbox"
+                    onChange={(_, isChecked) => setDeleteRPMmetadata(isChecked)}
+                    isChecked={deleteRPMmetadata}
+                />
+                <Checkbox label={_("Pending deployment")}
+                    key="pending-deployment"
+                    id="pending-deployment-checkbox"
+                    onChange={(_, isChecked) => setDeletePendingDeployments(isChecked)}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-403-20231124-085450-59247a42-fedora-coreos-devel//Coverage/lcov/github-pr.diff.gcov.html)